### DISCUSSION TITLE
What is the difference between Windows App SDK 0.8.6 and Windows App SDK 1.0.0?

### DISCUSSION THREAD

**Discussion by JamesLear92** [microsoft/WindowsAppSDK#1913] [Q&A] [answered]
Very confused, as Windows App SDK 0.8.6 was released after Windows App SDK 1.0.0? Which is the latest, and what is the difference between them?
Windows App SDK 0.8.6 boasts performance improvements, are these performance improvements also implemented in Windows App SDK 1.0.0?

**Comment by andrewleader** [comment, marked as answer]
Hi JamesLear92, good question. Version 0.8.6 is a **servicing update** to the 0.8.x version line. It's for customers that are still on 0.8 and don't (yet) want to update to 1.0 (updating to 1.0 requires a bit more work since it has a few breaking changes compared to the 0.8 API surface).

The latest version is 1.0.0 and that's what you should be using for new development. 0.8.6 is several features behind 1.0 (most notably, it doesn't support unpackaged apps, etc).

@BenJKuhn were the perf updates in 0.8.6 also included in 1.0.0? Or will those be added in a later 1.0.1 servicing update?

**Comment by BenJKuhn** [comment]
@andrewleader, The performance updates depended on a December update to .NET SDK (due to how the SDK picks up the reference to the C# WinRT runtime). We decided to keep the 1.0 release date as planned and roll those performance fixes into next servicing update for 1.0 instead.

Ben

**Reply by JamesLear92** [reply]
Just out of interest, what do these performance increases look like? One problem that a colleague and I have noticed with WinUI 3.0 on the SDK 1.0, is that it's locked at ~32 fps.

**Comment by JamesLear92** [comment]
Thank you very much both for the clarifications.

**Comment by BenJKuhn** [comment]
I'm not familiar with framerate concerns (the XAML & composition aspects of App SDK aren't my area of expertise). The improvements mostly relate to C# /WinRT projection changes. E.g. they substantially reduce the number of temporary managed objects created during interactions between C# & native code, reducing memory pressure and GC frequency. 

If you're interested in the finer details, this is representative of the type of work involved:
https://github.com/microsoft/CsWinRT/pull/1042

Most of the relevant changes are in the C# /WinRT repo over the last ~3 months:
https://github.com/microsoft/CsWinRT/pulls?q=is%3Apr+is%3Aclosed

Someone did report an issue related to framerate on the WinUI project here a few months ago, so at least there's an open issue tracking it:
https://github.com/microsoft/microsoft-ui-xaml/issues/6031

Ben
